addon/components/paper-button.js

```diff
@@ -7,14 +7,22 @@ import ColorMixin from 'ember-paper/mixins/color-mixin';
 const { computed } = Ember;
 
 export default BaseFocusable.extend(RippleMixin, ProxiableMixin, ColorMixin, {
-  attributeBindings: ['type'],
   type: 'button',
-  tagName: 'button',
+  tagName: computed('href', function() {
```

Comment:
I don't think it is *that* ugly. Init is the place to initialize variables after all.

tagName will nota be bound. You can't change tagName after render. That's the reason it can't be a computed. Did I get your question right?